packages/terra-date-input/src/DateInput.jsx

```diff
@@ -442,6 +468,30 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Sets state to new value dependent on existing date
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {Number} addDays Adds days to current date or today's date.
+   */
+  setHotKeyDate(event, addDays) {
+    let dateObj;
+    if (!this.state.month || !this.state.year || !this.state.day || addDays === 0) {
+      dateObj = new Date();
+    } else {
+      dateObj = new Date(`${this.state.year}-${this.state.month}-${this.state.day}`);
```

Comment:
What would happen if a month is selected (this.state.month is valid) but both `this.state.year` and `this.state.day` contain a value but they are invalid? For example: day is `6` and year is `202`. In this case it would get in here and create an invalid Date object to add `addDays`.

The logic might be easier if we check to ensure we have a full and complete date (month, day, and year are valid), then create the date based on these values and then add `addDays` to it. Otherwise, we don't have a complete date value and should fall back to use today's date.

Reply:
Added some logic for what constitutes a valid day and a valid year: https://github.com/cerner/terra-framework/pull/1483/commits/9d7f89c03340c4863ac691af55912ac4bd0c7e81